Net: Trivial-review: Make SendMessages easier to review

jtimon opened the pull request:
More things to do later with more disruption:

- The new functions can be turned into static unless someone else has a better idea
- The documentation should be moved to doxygen
- An indentation can be done for each of the new function
- MsgGetData() could be divided in 2, but I don't know if it makes sense

In the spirit of #5153, happy to go further dividing SendMessages in the low disruption territory, rem +13-3

jtimon committed:
Net: Trivial-review: Make SendMessages easier to review

More things to do later with more disruption:

- The new functions can be turned into static unless someone else has a better idea
- The documentation should be moved to doxygen
- An indentation can be done for each of the new function
- MsgGetData() could be divided in 2, but I don't know if it makes sense

JeremyRubin commented on src/net_processing.cpp:
```diff
@@ -2681,6 +2681,9 @@ class CompareInvMempoolOrder
     }
 };
 
+void MsgGetData(const Consensus::Params& consensusParams, CNode* pto, CConnman& connman, CNetMsgMaker& msgMaker, CNodeState& state, int64_t nNow, bool fFetch);
```
Maybe move the declarations (not definitions) to the header?

jtimon replied:
The idea was to leave them as static rather than expose them (unless someone else wants to expose some of them later). Perhaps I should have started from the top to avoid the extra declarations. No big deal for me to change it if we decide to divide the whole function ala #9579

JeremyRubin commented on src/net_processing.cpp:
```diff
@@ -2681,6 +2681,9 @@ class CompareInvMempoolOrder
 };
 
+void MsgGetData(const Consensus::Params& consensusParams, CNode* pto, CConnman& connman, CNetMsgMaker& msgMaker, CNodeState& state, int64_t nNow, bool fFetch);
+void MsgFeeFilter(CNode* pto, CConnman& connman, CNetMsgMaker& msgMaker);
```
I think CNetMsgMaker::Make could be const, which might be worth doing to be able to make this reference const.

jtimon replied:
No, CNetMsgMaker::Make() is not const :(

sipa replied:
I see no reason why it couldn't be const.

jtimon replied:
Right, I didn't realise both versions of CNetMsgMaker::Make can be just made const too.
Updating both this and #9608.

JeremyRubin reviewed (commented):
Concept ACK

rebroad commented on src/net_processing.cpp:
```diff
@@ -3116,7 +3119,14 @@ bool SendMessages(CNode* pto, CConnman& connman, std::atomic<bool>& interruptMsg
                 return true;
             }
         }
+        MsgGetData(consensusParams, pto, connman, msgMaker, state, nNow, fFetch);
```
Re nNow - Can't it work out the time by itself?

jtimon replied:
why call GetTimeMicros() more times than needed?

jtimon commented:
I would advice reviewers to look at  #9608 first, which I should have written first.

jtimon commented:
Rebased on top of #9659 which separates @JeremyRubin 's suggestion and includes a few more similar things.

jtimon commented:
Rebased after #9659 has been merged for clarity.

kallewoof commented on src/net_processing.cpp:
```diff
@@ -3164,7 +3167,14 @@ bool SendMessages(CNode* pto, CConnman& connman, const std::atomic<bool>& interr
+    return true;
+}
 
+void MsgGetData(const Consensus::Params& consensusParams, CNode* pto, CConnman& connman, const CNetMsgMaker& msgMaker, CNodeState& state, int64_t nNow, bool fFetch)
```
Maybe make `nNow` and `fFetch` const, to ensure they are not mistakenly updated (and discarded on return) in this method.

jtimon replied:
Long time ago, I remember I was asked not to do this for basic types. Recently we discussed a PR to change this in the whole codebase (sorry, I can't find it). But good good point, no reason not to do it here and in #9608. It also helps me read functions when more parameters are const.

kallewoof reviewed (commented):
utACK ab4f51e

jtimon commented:
Fixed nits ( https://github.com/bitcoin/bitcoin/pull/9608#issuecomment-283545547 https://github.com/bitcoin/bitcoin/pull/9579#discussion_r103590711 ) and rebased without needing it.

jtimon commented:
Rebased just because travis couldn't fetch a file.

jtimon commented:
Needed rebase, closing.
